### PR TITLE
[Minify] Initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
This PR adds initial integration of [Minify](https://github.com/tdewolff/minify). 

Minify is a Go package that implements minifiers for web formats (HTML, JS, CSS etc). It provides minifiers that are faster than existing solutions and it is widely used in the community. 

According to the author, the list of users includes but is not limited to:
 - Hugo (static website builder)
 - Caddy (popular alternative to Apache/NGINX)
 - CloudFlare (hosting and DNS solutions)
 - Beego (web framework in Go)
 - Gin (web framework in Go)
 - Google (internal team)
 - GitLab (internal team)

Tagging the author to keep him updated on the progress of the integration: @tdewolff

### CODE REVIEW

**Comment by tdewolff** [comment]
Thanks you @AdamKorcz for the effort! This will be of great help for the library, as I've used fuzzing in the past succesfully to fix numerous bugs. I assume that OSS-Fuzz pull in changes (from the library and the fuzz repository) automatically every now and then? Does it help to move the fuzz repository inside `minify`? Be aware by the way that the underlying `parse` library also contains fuzzing targets: https://github.com/tdewolff/fuzz/tree/master/parse, should we add those too?

**Comment by AdamKorcz** [comment]
@tdewolff Moving the fuzzing repository into the main one won't make a difference. We can consider it in case any problems arise, but the fuzzers all run fine as of now. And yes, updates to the branches in the build files should be considered in the fuzz-jobs.

Let's see if your good library is accepted, then we can definitely consider fuzzing the underlying `parse` library.

**Comment by tdewolff** [comment]
I've added the fuzz tests to `minify` anyways, I think it is better to keep everything as much in one repository as possible. Could you please update these lines in the build script @AdamKorcz ? Sorry for the inconvenience. See https://github.com/tdewolff/minify/tree/master/tests

> Let's see if your good library is accepted, and we can then definitely consider fuzzing the underlying parse library.

Agreed!

**Comment by inferno-chromium** [comment]
> I've added the fuzz tests to `minify` anyways, I think it is better to keep everything as much in one repository as possible. Could you please update these lines in the build script @AdamKorcz ? Sorry for the inconvenience. See https://github.com/tdewolff/minify/tree/master/tests
> 
> > Let's see if your good library is accepted, and we can then definitely consider fuzzing the underlying parse library.
> 
> Agreed!

We accept the project, @AdamKorcz , can you please build.sh and then we can merge.

**Comment by tdewolff** [comment]
Thanks for the work @AdamKorcz 

Should we incorporate https://github.com/tdewolff/parse in this PR, as it is quite tightly coupled with `minify`? It contains 5 small fuzzers in `tests/`. What do you think @inferno-chromium .

**Comment by inferno-chromium** [comment]
> Thanks for the work @AdamKorcz
> 
> Should we incorporate https://github.com/tdewolff/parse in this PR, as it is quite tightly coupled with `minify`? It contains 5 small fuzzers in `tests/`. What do you think @inferno-chromium .

Can just do this in another PR. Committing this for now.

File: projects/minify/build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  path=$1
+  function=$2
+  fuzzer=$3
+
+  go-fuzz -func $function -o $fuzzer.a $path
+
+  $CXX $CXXFLAGS $LIB_FUZZING_ENGINE $fuzzer.a -o $OUT/$fuzzer
+}
+
+find $GOPATH/src/github.com/tdewolff/minify/tests/*/ -maxdepth 1 -type d | while read target
```
Comment (tdewolff) [inline projects/minify/build.sh:27]: Shouldn't this me `-maxdepth 0`? It's showing me all subdirectories of each test case (i.e. suppressions, corpus, crashers, ...).

File: projects/minify/build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+do
+    fuzz_target=`echo $target | cut -d'/' -f 9`
```
Comment (tdewolff) [inline projects/minify/build.sh:29]: Does the `-f 9` always work? This depends on the length of `$GOPATH`, right? We could rewrite this as:

```bash
find $GOPATH/src/github.com/tdewolff/minify/tests/* -maxdepth 0 -type d | rev | cut -d'/' -f 1 | rev
```

Mind the lack of a `/` at the end of the `find` path. The `rev` allows to select the last item.

**Comment by inferno-chromium** [comment]
> Seems to be selecting `README.md` as a possible fuzzer? Do these comments have anything to do with that?

Sorry @tdewolff , should have waited on your review, can you please submit a PR for these fixes + those other 5 fuzzers enabling.

**Comment by AdamKorcz** [comment]
May I suggest moving the build script upstream? That way it will be easier for @tdewolff to work on updates?

**Comment by inferno-chromium** [comment]
> May I suggest moving the build script upstream? That way it will be easier for @tdewolff to work on updates?

Yes totally, go ahead, we actually recommend that in ideal integration - https://google.github.io/oss-fuzz/advanced-topics/ideal-integration/

**Comment by AdamKorcz** [comment]
Cool. @tdewolf FYI: The plan is to add the build script to the upstream repository and run that one when fuzzing through oss-fuzz. I will get the build script added asap.

**Pull request merged**